test/e2e/common/projected.go

```diff
@@ -0,0 +1,1518 @@
+												},
+											},
+											{
+												Path: "cpu_request",
+												ResourceFieldRef: &v1.ResourceFieldSelector{
+													ContainerName: "client-container",
+													Resource:      "requests.cpu",
+												},
+											},
+											{
+												Path: "memory_limit",
+												ResourceFieldRef: &v1.ResourceFieldSelector{
+													ContainerName: "client-container",
+													Resource:      "limits.memory",
+												},
+											},
+											{
+												Path: "memory_request",
+												ResourceFieldRef: &v1.ResourceFieldSelector{
+													ContainerName: "client-container",
+													Resource:      "requests.memory",
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			RestartPolicy: v1.RestartPolicyNever,
+		},
+	}
+
+	return pod
+}
+
+func applyLabelsAndAnnotationsToProjectedDownwardAPIPod(labels, annotations map[string]string, pod *v1.Pod) {
+	if len(labels) > 0 {
+		pod.Spec.Volumes[0].VolumeSource.Projected.Sources[0].DownwardAPI.Items = append(pod.Spec.Volumes[0].VolumeSource.Projected.Sources[0].DownwardAPI.Items, v1.DownwardAPIVolumeFile{
+			Path: "labels",
+			FieldRef: &v1.ObjectFieldSelector{
+				APIVersion: "v1",
+				FieldPath:  "metadata.labels",
+			},
+		})
+	}
+
+	if len(annotations) > 0 {
+		pod.Spec.Volumes[0].VolumeSource.Projected.Sources[0].DownwardAPI.Items = append(pod.Spec.Volumes[0].VolumeSource.Projected.Sources[0].DownwardAPI.Items, v1.DownwardAPIVolumeFile{
+			Path: "annotations",
+			FieldRef: &v1.ObjectFieldSelector{
+				APIVersion: "v1",
+				FieldPath:  "metadata.annotations",
+			},
+		})
+	}
+}
+
+func projectedAllVolumeBasePod(podName string, secretName string, configMapName string, labels, annotations map[string]string) *v1.Pod {
+	pod := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        podName,
+			Labels:      labels,
+			Annotations: annotations,
+		},
+		Spec: v1.PodSpec{
+			Volumes: []v1.Volume{
+				{
+					Name: "podinfo",
+					VolumeSource: v1.VolumeSource{
+						Projected: &v1.Projections{
+							Sources: []v1.VolumeProjection{
+								v1.VolumeProjection{
+									DownwardAPI: &v1.DownwardAPIVolumeSource{
+										Items: []v1.DownwardAPIVolumeFile{
+											{
+												Path: "podname",
+												FieldRef: &v1.ObjectFieldSelector{
+													APIVersion: "v1",
+													FieldPath:  "metadata.name",
+												},
+											},
+										},
+									},
+								},
+								v1.VolumeProjection{
+									Secret: &v1.SecretVolumeSource{
+										SecretName: secretName,
+									},
+								},
+								v1.VolumeProjection{
+									ConfigMap: &v1.ConfigMapVolumeSource{
+										LocalObjectReference: v1.LocalObjectReference{
+											Name: configMapName,
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			RestartPolicy: v1.RestartPolicyNever,
+		},
+	}
+
+	return pod
+}
```

Comment:
Add a test case that projects a secret, configmap, and downward API into the same volume